### PR TITLE
feature: configure API from DTL

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

Implement `configure` API based on [similar API in DTL](https://testing-library.com/docs/dom-testing-library/api-configuration).

Added initial option for [`asyncWaitTimeout`](https://testing-library.com/docs/dom-testing-library/api-configuration/#asyncutiltimeout).

### Test plan

* Added tests for `config` module functions
* Added tests for `waitFor` consuming `asyncWaitTimeout` option

### CODE REVIEW

**Review by AugustinLF — APPROVED** [review]

File: src/pure.ts [hunk 1]
```diff
@@ -16,9 +16,11 @@ export type {
   RenderResult as RenderAPI,
 } from './render';
 export type { RenderHookOptions, RenderHookResult } from './renderHook';
+export type { Config } from './config';
 
 export { act };
 export { cleanup };
+export { configure, resetToDefaults } from './config';
```
Comment (thymikee) [inline src/pure.ts:23]: Is `resetToDefaults` exported from DTL? Couldn't find it on their docs

**Review by thymikee — APPROVED** [review]
We could also implement `defaultHidden`, right? Looks good though

File: src/pure.ts (hunk 1 above)
Comment (mdjastrzebski) [reply src/pure.ts:23]: No it's not, only `configure` is, but I think it's good to have that for testinging convenience. Wdyt?

Comment (mdjastrzebski) [reply src/pure.ts:23]: Re `defaultHidden`: we first need to have https://github.com/callstack/react-native-testing-library/pull/1064 merged as I belive that `defaultHidden` is the default value of `hidden` query option, which we are yet to implement.

**Comment by mdjastrzebski** [comment]
re `defaultHidden`: we first need to merge https://github.com/callstack/react-native-testing-library/pull/1064, as it contains the `hidden` query options, and afaik `defaultHidden` is just the default value of that option.

**Comment by thymikee** [comment]
> re defaultHidden: we first need to merge https://github.com/callstack/react-native-testing-library/pull/1064, as it contains the hidden query options, and afaik defaultHidden is just the default value of that option.

Huh, it's opened for so long I thought it's already in 🙈

**Comment by mdjastrzebski** [comment]
It should be much easier done now when we have `isInaccessible` #1128  , which is the core logic behind `hidden` flag.

**Comment by thymikee** [comment]
Definitely! And please don't get me wrong, I'm not expecting to rush things here :)

**Pull request merged**